src/utils/desktop.ts

```diff
@@ -0,0 +1,50 @@
+import { getJson, postJson } from './net'
+
+interface DesktopStatus {
+  status: 0 | 1 | 2
+  address: string | null
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  config: Record<string, any>
+}
+
+export async function getDesktopStatus(): Promise<DesktopStatus> {
+  const response = await getJson(`http://${getDesktopHost()}/status`)
+
+  return response as DesktopStatus
+}
+
+export async function getBeeEthereumAddress(): Promise<string> {
+  const status = await getDesktopStatus()
+
+  if (!status.address) {
+    throw Error('Bee node Ethereum address is not yet available')
+  }
+
+  return status.address
+}
```

Comment:
This is a duplication, the Bee provider already has the ethereum address. I am fine with it, just wanted to mention this.

Reply:
Removed it, no longer needed